### PR TITLE
[terra-button] Enable visible focus when button is active

### PR DESCRIPTION
**Pull request by sugan2416**
### Summary

- Added visible focus to button when the active dom element is button
- This change was made to allow focus on button when `terra-menu` is closed

Closes #

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by sdadn** [comment]
Are there existing tests that cover the change in if statement?

File: packages/terra-button/src/Button.jsx [hunk 1]
```diff
@@ -173,7 +173,7 @@ class Button extends React.Component {
     }
 
     // Add focus styles for keyboard navigation
-    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN || document.activeElement === event.currentTarget) {
```
Comment (supreethmr) [inline packages/terra-button/src/Button.jsx:176]: I think event.currentTarget will always match document.active element !! Curious are there any scenario where event.currentTarget would not be equal to document.activeElement?? 

If event.currentTarget always matches document.active element then probably you should update this condition to 
```js
 ( document.activeElement === event.currentTarget && ( event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN ) )
```

Comment (sdadn) [reply packages/terra-button/src/Button.jsx:176]: Can some unit tests also be added to test the if statement and any potential edge cases?

Comment (sugan2416) [reply packages/terra-button/src/Button.jsx:176]: If we consider the case of terra-menu, upon Esc press, an opened menu closes and the programmatic focus is returned to button. This is triggering key event in button. If we update to suggested condition, it wouldn't pass as we didn't use return/tab key, thereby causing visible focus to fail in spite being the active element. Also mentioning that onfocus event isn't occurring in this case.

**Comment by supreethmr** [comment]
Closing this PR as changes has been made on terra-popup to return focus back onto terra-button on popup close :  https://github.com/cerner/terra-framework/pull/1641
